### DISCUSSION TITLE
IPC shared memory

### DISCUSSION THREAD

**Discussion by jdgriffith** [oven-sh/bun#4116] [Ideas]
Offer an easy and native way to share memory between processes. This would be more efficient than passing messages of data around.

**Comment by Jarred-Sumner** [comment]
Bun supports Bun.mmap which in theory could be used for this, but in practice it's still kind of difficult

**Reply by jdgriffith** [reply]
Oh nice! I will take a look. My understanding of shared memory is the process has to clean up after itself before it shuts down too but I could be wrong. It would be nice if the runtime cleaned up automatically.

Super excited about bun 1.0 btw. FWIW I don't think this feature idea is a 1.0 feature - probably something later.

**Comment by janwilmake** [comment]
I just remembered a couple years ago that I got stuck on shared memory (https://github.com/Karsens/pluribus-poker-bot-js/issues/2) with node.js 

It would be insanely cool if we could have this in Bun in Typescript
